packages/-ember-data/tests/integration/identifiers/lid-reflection-test.ts

```diff
@@ -293,4 +293,90 @@ module('Integration | Identifiers - lid reflection', function (hooks) {
     assert.deepEqual(cake.belongsTo('topping').id(), '2');
     assert.strictEqual(cake.topping.name, 'Cheese');
   });
+
+  test('child attributes are not updated correctly by .save() when sideposted record has different attribute than in the response payload with lid provided', async function (assert) {
+        class Topping extends Model {
+      @attr name;
+    }
+
+    class Cake extends Model {
+      @attr name;
+      @belongsTo('topping', { inverse: null, async: false }) topping;
+    }
+
+    this.owner.register('model:topping', Topping);
+    this.owner.register('model:cake', Cake);
+
+    class TestSerializer extends EmberObject {
+      normalizeResponse(_, __, payload) {
+        return payload;
+      }
+    }
+
+    class TestAdapter extends Adapter {
+      createRecord(_store, _ModelClass, snapshot) {
+        const lid = recordIdentifierFor(snapshot.record.topping).lid;
+
+        return resolve({
+          data: {
+            type: 'cake',
+            id: '1',
+            attributes: {
+              name: 'Cheesecake',
+            },
+            relationships: {
+              topping: {
+                data: {
+                  type: 'topping',
+                  id: '2',
+                  lid,
+                },
+              },
+            },
+          },
+          included: [
+            {
+              type: 'topping',
+              id: '2',
+              lid,
+              attributes: {
+                name: 'Cheese',
+              },
+              relationships: {
```

Comment:
if this relationships has is removed (or is made to include the lid of the cake) does the problem persist?